server/channels/app/channel.go

```diff
@@ -2970,57 +2970,31 @@ func (a *App) SearchChannelsUserNotIn(c request.CTX, teamID string, userID strin
 }
 
 func (a *App) MarkChannelsAsViewed(c request.CTX, channelIDs []string, userID string, currentSessionId string, collapsedThreadsSupported, isCRTEnabled bool) (map[string]int64, *model.AppError) {
-	// I start looking for channels with notifications before I mark it as read, to clear the push notifications if needed
-	channelsToClearPushNotifications := []string{}
-	if a.canSendPushNotifications() {
-		for _, channelID := range channelIDs {
-			channel, errCh := a.Srv().Store().Channel().Get(channelID, true)
-			if errCh != nil {
-				c.Logger().Warn("Failed to get channel", mlog.Err(errCh))
-				continue
-			}
+	var err error
 
-			member, err := a.Srv().Store().Channel().GetMember(context.Background(), channelID, userID)
-			if err != nil {
-				c.Logger().Warn("Failed to get membership", mlog.Err(err))
-				continue
-			}
+	user, err := a.Srv().Store().User().Get(c.Context(), userID)
+	if err != nil {
+		return nil, model.NewAppError("MarkChannelsAsViewed", "app.channel.update_last_viewed_at.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
```

Comment:
The translation code seems like a copy-paste error. Same for below as well.

Reply:
What this function is doing is updating the `lastViewedAt`, which is what the error states. I just followed the trend. Any suggestion on what other error should we show here? (keep in mind that any error coming from getting the user or the unreads and mentions is already bubbling up in the wrap.

Reply:
> I just followed the trend.

Yes, the trend is incorrect. `app.channel.update_last_viewed_at.app_error` should only be used for the error return from `a.Srv().Store().Channel().UpdateLastViewedAt`. All other actions should have unique translation codes.

> Any suggestion on what other error should we show here?

It should be a user readable message for what action failed - "Unable to get the user".